alternatives.go

```diff
@@ -0,0 +1,105 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+)
+
+// iptablesAlternativeSelector allows to configure a system to use iptables in
+// nft or legacy mode.
+type iptablesAlternativeSelector interface {
+	// useMode configures the system to use the selected iptables mode.
+	useMode(ctx context.Context, mode iptablesMode) error
+}
+
+// buildIptablesAlternativeSelector builds the proper iptablesAlternativeSelector depending
```

Comment:
I was considering `iptables` like one single word, since it's the name of a program. 

However, I do see there are already uses of `IPTables` in the k8s codebase so I'll change it for consistency. Thanks for pointing it!